components/builder-scheduler/src/data_store.rs

```diff
@@ -262,6 +261,43 @@ impl DataStore {
                            r#"CREATE OR REPLACE FUNCTION set_project_state_ident_v1 (pid bigint, jid bigint, state text, ident text) RETURNS void AS $$
                                   UPDATE projects SET project_state=state, job_id=jid, project_ident=ident, updated_at=now() WHERE id=pid;
                            $$ LANGUAGE SQL VOLATILE"#)?;
+
+        // Change the primary key for the groups table
+        migrator.migrate(
+            "scheduler",
+            r#"ALTER TABLE IF EXISTS groups DROP CONSTRAINT IF EXISTS groups_pkey;"#,
+        )?;
+        migrator.migrate(
+            "scheduler",
+            r#"CREATE SEQUENCE IF NOT EXISTS groups_id_seq;"#,
+        )?;
+        migrator.migrate("scheduler", r#"ALTER TABLE IF EXISTS groups ALTER COLUMN id SET DEFAULT next_id_v1('groups_id_seq');"#)?;
+        migrator.migrate(
+            "scheduler",
+            r#"ALTER TABLE IF EXISTS groups ADD PRIMARY KEY (id);"#,
+        )?;
+        migrator.migrate("scheduler",
+                     r#"CREATE OR REPLACE FUNCTION insert_group_v2 (
```

Comment:
If there's other code that's legitimately using the `v1` function, and the underlying data is compatible with both versions, then we should leave it (and probably make efforts to pull everything up to the new function ASAP). If nothing is using it, though, deletion makes a lot of sense, for the same reason that you'd delete Rust code that isn't being used.